src/main/java/algorithms/sorting/radixSort/README.md

```diff
@@ -0,0 +1,61 @@
+# Radix Sort
+
+Radix Sort is a non-comparison based, stable sorting algorithm. 
+
+Radix Sort continuously sorts based on the least-significant segment of a element 
+to the most-significant value of a element. 
+
+The definition of a 'segment' is user defined and defers from implementation to implementation. 
+Within our implementation, we define each segment as a bit chunk. 
+
+For example, if we aim to sort integers, we can sort each element 
+from the least to most significant digit, with the digits being our 'segments'. 
+
+While Radix Sort is non-comparison based, 
+the that total ordering of elements is still required. 
+This is maintained in how the states are stored after sorting is conducted with respect to each digit.
+
+This total ordering is needed because once we assigned a element to a order based on a segment, 
+the order *cannot* change unless deemed by a segment with a higher significance. 
+Hence a stable sort is required to maintain the order as 
+the sorting is done with respect to each of the segments.
+
+Within our implementation, we take the binary representation of the elements and 
+partition it into 8-bit segments, a integer is represented in 32 bits, 
+this gives us 4 total segments to sort through. 
+
+Note that the binary representation is weighted positional, 
+where each bit's value is dependent on its overall position 
+within the representation (the n-th bit from the right represents *2^n*), 
+hence we can actually increase / decrease the number segments we wish to conduct a split from.
+
+![Radix Sort](https://miro.medium.com/v2/resize:fit:661/1*xFnpQ4UNK0TvyxiL8r1svg.png)
+
+We place each element into a queue based on the number of possible segments that could be generated.
+Suppose the values of our segments are in base-10, (limited to a value within range *[0, 9]*), 
+we get 10 queues.
+
+*Source: Level Up Coding* 
+
+## Complexity Analysis
```

Comment:
i can help you a little here: 
in general, time complexity is

O((b/r)(n+2^r))
where b-bit word is broken into r-bit pieces and we apply counting sort b/r number of times.
the time complexity of counting sort is O(k+n) where n is the number of elements and k is the range (from 1 to k). here, since each piece is r-bit, range is 2^r.